client/src/components/Header/components/language-list.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useRef } from 'react';
+import type { TFunction } from 'i18next';
+import { connect } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+import { clientLocale } from '../../../../../config/env.json';
+import {
+  availableLangs,
+  LangNames,
+  LangCodes,
+  hiddenLangs
+} from '../../../../../config/i18n';
+import { hardGoTo as navigate } from '../../../redux/actions';
+import createLanguageRedirect from '../../create-language-redirect';
+import LanguageGlobe from '../../../assets/icons/language-globe';
+
+const locales = availableLangs.client.filter(
+  lang => !hiddenLangs.includes(lang)
+);
+
+const mapDispatchToProps = {
+  navigate
+};
+
+interface LanguageListProps {
+  t: TFunction;
+  navigate?: (location: string) => void;
+}
+
+export const LanguageList = ({
+  t,
+  navigate
+}: LanguageListProps): JSX.Element => {
+  const [showList, setShowList] = useState(false);
+  const listButtonRef = useRef<HTMLButtonElement>(null);
+
+  const handleClick = (): void => {
+    if (showList) {
+      setShowList(false);
+      return;
+    }
+    setShowList(true);
+  };
+
+  const handleLanguageChange = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ): void => {
+    const selectedLanguage = event.currentTarget.dataset.value;
+    const isSelecetedCurrentLanguage =
+      selectedLanguage === clientLocale || selectedLanguage === undefined;
+
+    event.preventDefault();
+    listButtonRef.current?.focus();
+    setShowList(false);
+    if (isSelecetedCurrentLanguage) {
+      return;
+    }
+    const path = createLanguageRedirect({
+      clientLocale,
+      lang: selectedLanguage
+    });
+    if (navigate) {
+      return navigate(path);
+    }
+  };
+
```

Comment:
```suggestion

const closeAndFocus = () => {
    listButtonRef.current?.classList.add('force-show');
    setShowList(false);
    setTimeout(() => {
      listButtonRef.current?.focus();
      listButtonRef.current?.classList.remove('force-show');
    }, 100);
  }
  
```

When using the `Esc` key to close the menu I'm noticing that the screen reader will announce that the menu is still expanded even if we set `aria-expanded` to `false` before setting focus on the button. I think the accessibility tree just needs a little more time to pick up the change. This function allows us to set `aria-expanded` to `false` and then delay just a bit before setting focus on the button. We need the `force-show` class to keep the menu open when `aria-expanded` is set to `false` otherwise the menu will be `display: none` and we'll lose keyboard focus before we set it on the button.